src/lib/core/TLVWriter.cpp

```diff
@@ -290,12 +290,26 @@ CHIP_ERROR TLVWriter::PutBytes(Tag tag, const uint8_t * buf, uint32_t len)
 
 CHIP_ERROR TLVWriter::PutString(Tag tag, const char * buf)
 {
-    size_t len = strlen(buf);
+    if (buf == nullptr)
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    else if (mMaxLen == 0)
+        return CHIP_ERROR_INCORRECT_STATE;
+
+    // No length is provided, so we cannot assume a well-formed C string.
+    // Use the container size as the limit when dereferencing the buffer.
```

Comment:
Probably worth documenting that this is not going to prevent issues with running off the end of `buf` if `buf` is not in fact null-terminated.  All it might do is limit how much we run off, and hence limit the chance we run into non-readable memory and crash.  The heap-memory information leak would still be there.

Reply:
True, added a disclaimer.